hazelcast/src/main/java/com/hazelcast/kubernetes/KubernetesApiProvider.java

```diff
@@ -58,7 +58,7 @@ default Integer extractPort(JsonValue subsetJson) {
                 }
             }
         }
-        if (ports.size() == 1) {
+        if (ports.size() > 0) {
```

Comment:
Is there a reason why changed the size check here?

Reply:
The reason is that for the advanced network setup, it will return a `null` port. And although it will work when the `port` is specified in the config, without it specified it will still be `null`

Reply:
But when I check the code I see we only care about the first port. Is this the way we want to go with this?

Reply:
Ideally, not, but it requires the changes in the core API so that one endpoint can have separate ports for different protocols. Until this change is done, the whole part is handled by the core that is returning the port configured in the  `advanced-network` section

Reply:
I have an additional question, do we depend on some port being the first element in ports list?

Reply:
No, as with the advanced network enabled, the Hazelcast will override the port based on the socket protocol. For the other cases, it is recommended to set the `hazelcast-service-port` port name

Reply:
@SeriyBg Hi, note on this.  This change appears to be breaking our deployment with 5.3.0.  The port name `hazelcast-service-port` is too long for k8s port names, so the check never matches.  In k8s we have:
```json
      "ports": [
        {
          "name": "http",
          "port": 8443,
          "protocol": "TCP"
        },
        {
          "name": "admin",
          "port": 8444,
          "protocol": "TCP"
        },
        {
          "name": "hazelcast",
          "port": 5701,
          "protocol": "TCP"
        }
      ]
```
It does not find the `hazelcast` port.